### PR TITLE
Use smaller and compressed varients of buttons and form components 

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
Replace instances of `EuiButton` that don't have an explicit sizing attribute to `EuiSmallButton*`.
Replace instances of `Eui<form elements>` that don't have density attributes to `EuiCompressed<form elements>`.

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
- [X] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/workflow_detail/workflow_inputs/input_fields/model_field.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ export function ModelField(props: ModelFieldProps) {
       <Field name={props.fieldPath}>
         {({ field, form }: FieldProps) => {
           return (
-            <EuiFormRow
+            <EuiompressedFormRow
```
Comment (ohltyler) [inline public/pages/workflow_detail/workflow_inputs/input_fields/model_field.tsx:88]: Missing `C`. Same on lines 99 / 138

**Review by ohltyler — COMMENTED** [review]
Thanks @AMoo-Miki! Just one small syntax comment.

Btw, is there any standardized guidelines for this listed somewhere I can reference for the future?

File: public/pages/workflow_detail/workflow_inputs/input_fields/model_field.tsx (hunk 1 above)
Comment (AMoo-Miki) [reply public/pages/workflow_detail/workflow_inputs/input_fields/model_field.tsx:88]: thanks

**Comment by AMoo-Miki** [comment]
> Btw, is there any standardized guidelines for this listed somewhere I can reference for the future?

Viraj owns it. Can't tag him here somehow.

**Review by ohltyler — APPROVED** [review]

**Pull request merged**